outlines/caching.py

```diff
@@ -56,3 +56,11 @@ def clear_cache():
     """Erase the cache completely."""
     global memory
     memory.storage.clear()
+
+
+def enable_cache():
+    """Enable the cache."""
+    global memory
+
+    if isinstance(memory, NoCache):
+        memory = Cache(serializer=JSONSerializer(), storage=LocalFileStorage(cache_dir))
```

Comment:
I left this in from when I was attempting to test the seeding with and without caching in `tests/models/test_hf_transformers.py`; however, it looks like the caching approach might be restricting this sort of dynamic on/off functionality, because the cached [`call_model_generate_method`](https://github.com/normal-computing/outlines/blob/2af0b5ab05182ae86fdd931dd463b1c6dd986136/outlines/models/hf_transformers.py#L65) is constructed at module load time.  If we construct `call_model_generate_method` or apply the `cache` decorator to it [when completers `call` functions are constructed](https://github.com/normal-computing/outlines/blob/2af0b5ab05182ae86fdd931dd463b1c6dd986136/outlines/models/hf_transformers.py#L49), we might be able to remove that restriction.